cdi-core/src/main/java/com/linkedin/cdi/extractor/JsonExtractor.java

```diff
@@ -583,7 +583,7 @@ private JsonObject retrievePushDowns(JsonElement response, Map<String, Map<Strin
     for (Map.Entry<String, Map<String, String>> entry : derivedFields.entrySet()) {
       String source = entry.getValue().get("source");
       if (JsonUtils.has(data, source)) {
-        pushDowns.addProperty(entry.getKey(), data.get(source).getAsString());
```

Comment:
Earlier when testing with `ms.derived.fields` I configured it as `'[{"name": "key", "formula": {"type": "string", "source": "key1.key2"}}]'`. The source JsonObject `obj` contains the `key1.key2` subfield but I still get a NullPointerException. I think this may be due to a bug at line #586 where we should have used `JsonUtils.get` to retrieve fields from a nested Json.